pkg/plugin/plugin.go

```diff
@@ -0,0 +1,119 @@
+// Copyright 2018-2021 CERN
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// In applying this license, CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+package plugin
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/go-plugin"
+)
+
+// RevaPlugin represents the runtime plugin
+type RevaPlugin struct {
+	Plugin interface{}
+	Client *plugin.Client
+}
+
+// Kill kills the plugin process
+func (plug *RevaPlugin) Kill() {
+	plug.Client.Kill()
+}
+
+var handshake = plugin.HandshakeConfig{
+	ProtocolVersion:  1,
+	MagicCookieKey:   "BASIC_PLUGIN",
+	MagicCookieValue: "hello",
+}
+
+func compile(pluginType string, path string) (string, error) {
+	wd, err := os.Getwd()
+	if err != nil {
+		return "", fmt.Errorf("could not find current directory: %v", err)
+	}
+	binaryPath := filepath.Join(wd, "bin", pluginType, filepath.Base(path))
```

Comment:
We need to create this directory. Also, not sure about creating it in the directory where you're running reva from. Maybe `/var/tmp/reva/bin/`?

Reply:
Okay no, this works. I only ran into the other issue

Reply:
Done!